frontend/utils/nativeNotification.js

```diff
@@ -0,0 +1,27 @@
+function checkNotification (): boolean {
+  return !!window.Notification
+}
+
+export async function requestNotificationPermission (force: boolean = false): Promise<null | string> {
+  if (!checkNotification() ||
+    (!force && Notification.permission !== 'default') ||
+    (force && Notification.permission !== 'granted')) {
```

Comment:
There seem to be two problems here:

1. It's unclear to me what `force` means here (it should be added in a comment what it does), but I assume it's intended to be used to re-request permissions even if they were previously not granted. If that's the case, this logic is wrong, since if `force = true` and `Notification.permission !== 'granted'` we want to proceed.
2. I see nowhere else in the code where `force = true`. It seems this variable isn't used. And yet it should be used in the settings.

Reply:
1. It is necessary when user wants to request permission if they have previously denied. You are right. Nice catch.
2. We don't call this function with `force` parameter right now. But if we have design to request permission, we will use this parameter.

Reply:
> We don't call this function with force parameter right now. But if we have design to request permission, we will use this parameter.

Well we do have a design: that's how the current behavior of the Notifications settings should work. It should pass in `force: true` so that if the user rejected notifications initially, they can re-request them

Reply:
Yeah. Just checked. Design is ready and UI implemented is missing.

Reply:
I think we should use `force: true` parameter only in the `NotificationSettings` page.
Right now, we request permission when user do mousemove in dashboard.
What do you think, @taoeffect?

Reply:
So we should implement NotificationSettings page first, and call this function passing `force: true` parameter.

Reply:
> I think we should use force: true parameter only in the NotificationSettings page.

Right, that's what I was saying, that the NotificationSettings should use this, to allow users to re-request permissions if they initially said no.

Reply:
Should I implement NotificationSettings page and call this function passing `force: true` in this PR?

Reply:
Yes, that's what I've been saying, lol. You already have the link in NotificationSettings that says `Request Permission`. You just need to pass in `true` for `force` there, and fix the logic error here